pkg/engine/context/loaders/apicall.go

```diff
@@ -54,7 +54,7 @@ func (a *apiLoader) LoadData() error {
 	}
 	if a.data == nil {
 		var err error
-		if a.data, err = executor.Fetch(a.ctx); err != nil {
+		if a.data, err = executor.Fetch(a.ctx); err != nil && a.entry.APICall.Default == nil {
```

Comment:
you dont have to check `a.entry.APICall.Default == nil` value here anymore

Reply:
If Fetch returns error then a.data will be nil in that case there arises 2 conditions. 1. is default is present then we will use default. 2. If default is not given by user then we need to return error. If I remove this `a.entry.APICall.Default == nil` it wont consider default when Fetch returns error.